app/javascript/Pages/Signup.jsx

```diff
@@ -12,9 +11,7 @@ export default function Login(props) {
   };
 
   const handleSubmit = (e) => {
-    const token = document.querySelector(
-      "[name=csrf-token]"
-    ).content;
+    const token = document.querySelector("[name=csrf-token]").content;
```

Comment:
**suggestion (blocking)**: This piece of code is needed in almost all components. You can use React Context to only run this once and access it anywhere you need it. Here is a good example of how to set it up https://ibaslogic.com/react-context-api/